### PR TITLE
fix(output.datadog): log response in case of non 2XX response from API

### PR DESCRIPTION
**Pull request by neelayu**
# Required for all PRs

- [ ] Updated associated README.md.
- [ ] Wrote appropriate unit tests.
- [ ] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves: NA

Add a debug log to print the response string obtained from API in case of a non 2XX status.

### CODE REVIEW

**Comment by neelayu** [comment]
!signed-cla

File: plugins/outputs/datadog/datadog.go [hunk 1]
```diff
@@ -168,6 +169,9 @@ func (d *Datadog) Write(metrics []telegraf.Metric) error {
 	defer resp.Body.Close()
 
 	if resp.StatusCode < 200 || resp.StatusCode > 209 {
+		// err can be ignored
+		body, _ := io.ReadAll(resp.Body)
+		d.Log.Debugf("Response from datadog api %s", string(body))
 		return fmt.Errorf("received bad status code, %d", resp.StatusCode)
```
Comment (powersj) [inline plugins/outputs/datadog/datadog.go:175]: if we are already returning an error, is there a reason to not add the body to that error message? Something like:

```suggestion
		return fmt.Errorf("received bad status code, %d: %s", resp.StatusCode, string(body))
```

Comment (neelayu) [reply plugins/outputs/datadog/datadog.go:175]: Although that is a good option, I left it out from the error message considering the body could be large string which would also increase the error message size. If that is acceptable, then we can make that change.

Comment (powersj) [reply plugins/outputs/datadog/datadog.go:175]: Do you have an example of a really long message you could share?
Where we say bad status code in other places, we have been slowly adding the body in as well, so I think it should be ok, but what a long message looks like might help. 

Thanks!

Comment (neelayu) [reply plugins/outputs/datadog/datadog.go:175]: Official documents dont have long messages, but we have a usecase where we sort of validate the datadog json body at a proxy and forward it to the cloud. Our validation could return a json which is around 20 lines. But given that it is a specific usecase, I feel its acceptable to use what you suggest. 
Thanks!

**Review by powersj — APPROVED** [review]

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for your contribution @neelayu!

**Pull request merged**